tests/playwright/sanity/core/app/modules/onboarding/onboarding.utils.js

```diff
@@ -0,0 +1,438 @@
+    }
+
+    async checkUserIsOnStepTwo() {
+        await expect( this.page.url(), `User is not on Step 2 but instead on ${ await this.page.url() }` ).toEqual( this.baseUrl + this.step2URL );
+    }
+
+    async selectContinueWithHelloThemeButton() {
+        await this.continueWithHelloThemeButton.click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.waitForTimeout( 250 );
+    }
+
+    async checkUserIsOnStepThree() {
+        await expect( this.page.url(), `User is not on Step 3 but instead on ${ await this.page.url() }` ).toEqual( this.baseUrl + this.step3URL );
+    }
+
+    async checkElementorThemeIsActive() {
+        // Await this.page.waitForSelector( '#hello-elementor-name span' );
+        await expect( await this.elementorTheme.getAttribute( 'class' ), `Elementor Theme is not active` ).toEqual( 'theme active' );
+    }
+
+    async checkTwenty21ThemeIsActive() {
+        await expect( await this.twenty21Theme.getAttribute( 'class' ), `The Twenty 21 Theme is not active` ).toEqual( 'theme active' );
+    }
+
+    async checkDisclaimerIsPresent( disclaimer ) {
+        await expect( this.disclaimerNotice ).toContainText( disclaimer );
+    }
+
+   /*
+    * Third Step - Site Name - Functions
+    */
+    async gotoStep3() {
+        await this.page.goto( this.step3URL );
+    }
+
+    async extractSiteTitle() {
+        return await this.page.evaluate( () => elementorAppConfig.onboarding.siteName );
+    }
+
+    async checkTitleIsCorrect( title ) {
+        await this.page.waitForSelector( this.siteTitleFieldSelector );
+        await expect( await this.siteTitleField.getAttribute( 'value' ) ).toEqual( title );
+    }
+
+    async emptySiteNameField() {
+        await this.page.fill( 'input[type="text"]', '' );
+    }
+
+    async insertPageName( pageName ) {
+        await this.page.fill( 'input[type="text"]', pageName );
+    }
+
+   /*
+    * Fourth Step - Site Logo - Functions
+    */
+    async gotoStep4() {
+        await this.page.goto( this.step4URL, { waitUntil: 'networkidle' } );
+    }
+
+    async checkStepFourURL( url ) {
+        await expect( await this.page.url(), `User is not on Step 4 but instead on ${ await this.page.url() }` ).toEqual( this.baseUrl + url );
+    }
+
+    async goToSiteItentityPage() {
+        await this.page.goto( this.customizePage, { waitUntil: 'networkidle' } );
+        await this.page.locator( '#accordion-section-title_tagline' ).click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.waitForSelector( '#customize-control-custom_logo .upload-button' );
+    }
+
+    async uploadLogo() {
+        await this.page.locator( '#customize-control-custom_logo .upload-button' ).first().click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.locator( '#menu-item-browse' ).click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.waitForTimeout( 500 );
```

Comment:
Can you compare the upload process of images e.g. for the Image Carousel or Image Gallery, to see it we can find a way to avoid the timeout?